internal/super/setup.go

```diff
@@ -77,134 +77,163 @@ func create(
 	var err error
 
 	if setupFlags.Scaffold || setupFlags.ScaffoldID != 0 {
-		targetDir, err = getTargetDirectory(args[0])
-		if err != nil {
-			return nil, err
+		// Error if no project name is given
+		if len(args) < 1 || args[0] == "" {
+			return nil, fmt.Errorf("no project name provided")
 		}
 
-		selectedScaffold, err := selectScaffold(logger)
+		targetDir, err = handleScaffold(args[0], logger)
 		if err != nil {
-			return nil, fmt.Errorf("error selecting scaffold %w", err)
-		}
-
-		logger.StartProgress(fmt.Sprintf("Creating your project %s", targetDir))
-		defer logger.StopProgress()
-
-		if selectedScaffold != nil {
-			err = cloneScaffold(targetDir, *selectedScaffold)
-			if err != nil {
-				return nil, fmt.Errorf("failed creating scaffold %w", err)
-			}
+			return nil, err
 		}
 	} else {
-		// Ask for project name if not given
-		if len(args) < 1 {
-			userInput, err := prompt.RunTextInput("Enter the name of your project", "Type your project name here...")
-			if err != nil {
-				fmt.Printf("Error running project name: %v\n", err)
-				os.Exit(1)
-			}
-
-			targetDir, err = getTargetDirectory(userInput)
-			if err != nil {
-				return nil, err
-			}
-		} else {
-			targetDir, err = getTargetDirectory(args[0])
-			if err != nil {
-				return nil, err
-			}
-		}
-
-		params := config.InitConfigParameters{
-			ServiceKeySigAlgo:  "ECDSA_P256",
-			ServiceKeyHashAlgo: "SHA3_256",
-			Reset:              false,
-			Global:             false,
-			TargetDirectory:    targetDir,
-		}
-		state, err := config.InitializeConfiguration(params, logger, state.ReaderWriter())
+		targetDir, err = startInteractiveSetup(args, logger, state)
 		if err != nil {
-			return nil, fmt.Errorf("failed to initialize configuration: %w", err)
+			return nil, err
 		}
+	}
 
-		// Generate standard cadence files
-		// cadence/contracts/DefaultContract.cdc
-		// cadence/scripts/DefaultScript.cdc
-		// cadence/transactions/DefaultTransaction.cdc
-		// cadence/tests/DefaultContract_test.cdc
+	return &setupResult{targetDir: targetDir}, nil
+}
 
-		directoryPath := filepath.Join(targetDir, "cadence")
+func startInteractiveSetup(
+	args []string,
+	logger output.Logger,
+	state *flowkit.State,
+) (string, error) {
+	var targetDir string
+	var err error
 
-		_, err = generateNew([]string{"DefaultContract"}, "contract", directoryPath, logger, state)
+	// Ask for project name if not given
+	if len(args) < 1 {
+		userInput, err := prompt.RunTextInput("Enter the name of your project", "Type your project name here...")
 		if err != nil {
-			return nil, err
+			fmt.Printf("Error running project name: %v\n", err)
+			os.Exit(1)
```

Comment:
```suggestion
			return nil, fmt.Errorf("Error running project name: %v", err)
```

Is there a reason we can't use the standard error procedure?